### PR TITLE
JIT Pinning support for JDK19 on Power

### PR DESCRIPTION
**Pull request by IBMJimmyk**
This commit adds changes to Power to support Virtual Threads.

This includes incrementing and decrementing ownedMonitorCount when locks are acquired or released.

Also callOutCount is incremented and decremented on JNI call outs.

### CODE REVIEW

**Comment by IBMJimmyk** [comment]
I added a bunch of comments. I also finished running tests to confirm that the expected assembly is being generated for each of the locking and unlocking paths and also JNI call outs.

**Comment by IBMJimmyk** [comment]
I think everything looks good now. @zl-wang Can you take a look at this?

File: runtime/compiler/p/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -4771,13 +4776,23 @@ reservationLockEnter(TR::Node *node, int32_t lwOffset, TR::CodeGenerator *cg, TR
...
-   generateDepLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, conditions);
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel);
+#if JAVA_SPEC_VERSION >= 19
+      /*
+       * Adjust J9VMThread ownedMonitorCount for execution paths that do not
+       * go out of line to acquire the lock. It is safe and efficient to do
+       * this unconditionally. There is no need to check for overflow.
+       */
+      generateTrg1MemInstruction(cg, TR::InstOpCode::Op_load, node, tempReg, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset(), TR::Compiler->om.sizeofReferenceAddress()));
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, tempReg, tempReg, 0x1);
```
Comment (zl-wang) [inline runtime/compiler/p/codegen/J9TreeEvaluator.cpp:4787]: general comment:  you are using 0x1 or -0x1 in the code ... that just looks strange to me. is this a particular coding style you are following?

File: runtime/compiler/p/codegen/J9TreeEvaluator.cpp [hunk 2]
```diff
@@ -4771,13 +4776,23 @@ reservationLockEnter(TR::Node *node, int32_t lwOffset, TR::CodeGenerator *cg, TR
...
+      generateMemSrc1Instruction(cg, TR::InstOpCode::Op_st, node, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset(), TR::Compiler->om.sizeofReferenceAddress()), tempReg);
+#endif //JAVA_SPEC_VERSION >= 19
 
    TR_PPCOutOfLineCodeSection *outlinedHelperCall = new (cg->trHeapMemory()) TR_PPCOutOfLineCodeSection(node, TR::call, NULL, callLabel, doneOOLLabel, cg);
    cg->getPPCOutOfLineCodeSectionList().push_front(outlinedHelperCall);
 
-   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneOOLLabel);
+   generateDepLabelInstruction(cg, TR::InstOpCode::label, node, doneOOLLabel, conditions);
```
Comment (zl-wang) [inline runtime/compiler/p/codegen/J9TreeEvaluator.cpp:4794]: done00LLable might reduce the code readability.  in later code, i saw callReturnLabel which sounds better and more readable.

File: runtime/compiler/p/codegen/J9TreeEvaluator.cpp (hunk 1 above)
Comment (IBMJimmyk) [reply runtime/compiler/p/codegen/J9TreeEvaluator.cpp:4787]: No particular reason so I can change it.

File: runtime/compiler/p/codegen/J9TreeEvaluator.cpp (hunk 2 above)
Comment (IBMJimmyk) [reply runtime/compiler/p/codegen/J9TreeEvaluator.cpp:4794]: I can change this.

**Review by zl-wang — APPROVED** [review]
Looks good ...

**Pull request merged**